contracts/RelayHub.sol

```diff
@@ -179,46 +166,44 @@ contract RelayHub is IRelayHub {
     }
 
     struct RelayCallData {
-        uint256 initialGas;
+        bool success;
         bytes4 functionSelector;
         bytes recipientContext;
         IPaymaster.GasLimits gasLimits;
         RelayCallStatus status;
     }
+
     function relayCall(
     // TODO: msg.sender used to be treated as 'relay' (now passed in a struct),
     //  make sure this does not have security impl
         ISignatureVerifier.RelayRequest calldata relayRequest,
         bytes calldata signature,
-        bytes calldata approvalData
+        bytes calldata approvalData,
+        uint externalGasLimit
```

Comment:
Probably cannot be left unconstrained. Will add require(externalGasLimit  < blockGasLimit) to prevent some relay draining more than it's stake

Reply:
how can it drain anything? it doesn't really matter, as far as the EVM cares. its the real gas limit that determine the upper bound.

Reply:
I am a relay server
I set externalGasLimit = 1e20 or whatever

uint256 gasUsed = (externalGasLimit - gasleft()) + GAS_OVERHEAD

charge will be enormous

Reply:
indeed. needs `require(externalGasLimit<tx.gaslimit)` (upper "attack" limit near 8M gas)
(I wanted `require(externalGasLimit-gasleft()<val)`, but we depend on current calldata pricing)

Reply:
Done